src/registers/model_specific.rs

```diff
@@ -172,4 +203,125 @@ mod x86_64 {
             unsafe { Self::MSR.write(address.as_u64()) };
         }
     }
+
+    impl Star {
+        /// Read the Ring 0 and Ring 3 segment bases.
+        /// The remaining fields are ignored because they are
+        /// not valid for long mode.
+        ///
+        /// # Returns
+        /// - Field 1 (SYSRET): The CS selector is set to this field + 16. SS.Sel is set to
+        /// this field + 8. Because SYSRET always returns to CPL 3, the
+        /// RPL bits 1:0 should be initialized to 11b.
+        /// - Field 2 (SYSCALL): This field is copied directly into CS.Sel. SS.Sel is set to
+        ///  this field + 8. Because SYSCALL always switches to CPL 0, the RPL bits
+        /// 33:32 should be initialized to 00b.
+        pub fn read_raw() -> (u64, u64) {
+            let msr_value = unsafe { Self::MSR.read() };
+            let sysret = msr_value.get_bits(48..63);
+            let syscall = msr_value.get_bits(32..47);
```

Comment:
`..` denotes an _exclusive_ range, so I think you're accidentally ignoring the highest level bit of each field here.